nativelib/src/main/scala/scala/re2s/UnicodeTableData.scala

```diff
@@ -0,0 +1,2572 @@
+package scala.re2s
+
+trait UnicodeTableData {
+  private val _Lm: Array[Array[Int]] =
+    Array(
+      Array(0x02b0, 0x02c1, 1),
```

Comment:
Understood, time to open the black box and apply what you have taught me (my words, not yours).  This is pending depending upon what I can figure out about common data with
Character.scala.

Reply:
Another option here is to store in "column major format" if it makes the code any simpler/easier to read than computing indexes into a giant single dimensional array that contains all the data. For example, instead of:
```
val myarr = Array(
  Array(a1, b1, c1),
  Array(a2, b2, c2),
  ...
```
We can do:
```
val myarrA = Array(
  a1,
  a2,
  ...
)
val myarrB = Array(
  b1,
  b2
  ...
)
val myarrC = Array(
  c1,
  c2
  ...
)
```

Reply:
Good suggestion. Parallel arrays make the code easier to both to write and get correct
and to read later on.

I am currently working on ripping re2c out.  With your agreement, I'll work on this
after rec2 has seen the door.

Reply:
This is the approach I took based on @sjrd 's help for upper and lower case. The range is searched via [binary search here](https://github.com/scala-native/scala-native/blob/master/javalib/src/main/scala/java/lang/Character.scala#L1455) and then the other arrays are to calculate offsets etc. as explained starting [here.](https://github.com/scala-native/scala-native/blob/master/javalib/src/main/scala/java/lang/Character.scala#L1215)

I reviewed `re2s` a bit and see they are doing something similar although I would need to study harder to really understand exactly what they are doing. The nested arrays are an interesting approach but I'm not sure why that approach was taken. It would be great to unify the usage of unicode as then we code generate new arrays based on a new unicode version. Sharing the data in runtime is a really good idea as well.

Reply:
@ekrich It really seems like unicode unification should be a stand-alone PR. It doesn't seem to be trivial and it distracts from the main goal of the current PR: getting rid of old regex implementation.

Reply:
@LeeTibbert Yes, lets focus on getting rid off c-based re2 in this PR.

Reply:
@densh and @LeeTibbert I totally agree, just wanted to affirm my support for your ideas.